provers/stark/src/examples/bit_flags.rs

```diff
@@ -45,11 +44,22 @@ impl TransitionConstraint<StarkField, StarkField> for BitConstraint {
 
     fn evaluate(
         &self,
-        frame: &Frame<StarkField, StarkField>,
-        transition_evaluations: &mut [Felt252],
-        _periodic_values: &[Felt252],
-        _rap_challenges: &[Felt252],
+        evaluation_context: &TransitionEvaluationContext<StarkField, StarkField>,
+        transition_evaluations: &mut [FieldElement<StarkField>],
     ) {
+        let (frame, _periodic_values, _rap_challenges) = match evaluation_context {
+            TransitionEvaluationContext::Prover {
+                frame,
+                periodic_values,
+                rap_challenges,
+            }
+            | TransitionEvaluationContext::Verifier {
+                frame,
+                periodic_values,
+                rap_challenges,
+            } => (frame, periodic_values, rap_challenges),
+        };
```

Comment:
could this pattern be made into a method of the `TransitionEvaluationContext` maybe? I know the generics may bother the compiler so if this is troublesome then nevermind, we can think of it later.

Reply:
yeah I suspected it, not really important to do right now